pkg/sql/pg_catalog_diff.go

```diff
@@ -115,8 +115,10 @@ func (p PGCatalogTables) isDiffOid(
 		return true
 	}
 
+	// For columns that are expected to be missing, the diff is stored as nil
+	// and is present in the map.
 	diff, ok := columns[columnName]
-	if !ok {
+	if !ok || diff == nil {
```

Comment:
Wait, if the column is expected to be missing and is stored as nil, shouldn't we return false so it is ignored in `pg_catalog_test.go` line 204?

Reply:
well if we return false, then line 204 will skip over this, so the test will pass when it should not.

ah actually, the root issue is that in general, we do want to allow crdb to have extra columns that are not in postgres, since sometimes we add additional columns for other reasons. so this change here will make it so that if we do add a column that overlaps with a postgres column, then it will check that it's an expected OID.

Reply:
Ah I thought the intent was for the test to "pass" if it was an expected difference.

>so this change here will make it so that if we do add a column that overlaps with a postgres column, then it will check that it's an expected OID.

Gotcha

Reply:
oh yeah i think this counts as an unexpected difference so it should fail

the reason i added this is that without this check, the test would fail with a nil pointer error.